include/seqan/bam_io/cigar.h

```diff
@@ -149,14 +149,14 @@ uint32_t toBamCigarElement(CigarElement<TOperation, TCount> const & cigarElement
 {
     char operation = 0;
     switch (cigarElement.operation) {
-        case 'X': operation += 1;
-        case '=': operation += 1;
-        case 'P': operation += 1;
-        case 'H': operation += 1;
-        case 'S': operation += 1;
-        case 'N': operation += 1;
-        case 'D': operation += 1;
-        case 'I': operation += 1;
+        case 'X': operation += 1; SEQAN_FALLTHROUGH;
+        case '=': operation += 1; SEQAN_FALLTHROUGH;
+        case 'P': operation += 1; SEQAN_FALLTHROUGH;
+        case 'H': operation += 1; SEQAN_FALLTHROUGH;
+        case 'S': operation += 1; SEQAN_FALLTHROUGH;
+        case 'N': operation += 1; SEQAN_FALLTHROUGH;
+        case 'D': operation += 1; SEQAN_FALLTHROUGH;
+        case 'I': operation += 1; SEQAN_FALLTHROUGH;
         case 'M': break;
     }
     return (cigarElement.count << 4) | operation;
```

Comment:
Wow, this looks really bad, I have no clue what the intended behaviour should be.
I just checked and this function is
* not documented
* not used anyway

I would propose adding the FALLTHROUGHs, but also adding a deprecated notice saying that the behaviour is not clear and that this function should not be used anymore.